include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -110,32 +109,82 @@ class pool_memory_resource final
   friend class detail::stream_ordered_memory_resource<pool_memory_resource<Upstream>,
                                                       detail::coalescing_free_list>;
 
+  /**
+   * @brief Construct a `pool_memory_resource` and allocate the initial device memory
+   * pool using `upstream_mr`.
+   *
+   * @deprecated Use the constructor that takes an explicit initial pool size instead.
```

Comment:
It seems impossible to link to overloaded templated constructors in doxygen, so I was not able to do that.